Portfolio Hand-in

carro-barro opened the pull request:
netlify: https://carolina-oldertz-portfolio.netlify.app/

figma: https://www.figma.com/design/YFJswIB708jYUoHg3swht2/Portfolio-designs--Copy-?node-id=1-1339&m=dev

note: i took away some of the socials because i don't use them

Commit:
started to divide the hero part into components and styled them using styled components

Commit:
continued with the hero section and made the tech section and started with the projects section, i added some info in the json file about the projects

Commit:
deleted the different hero image files and made a function that changes the styling with props, started to loop the project.json and display it in the browser, but needs more styling

Commit:
started with the skillsection and articlesection, need more styling and figuring out the logic, also made a article json and started to add info

Commit:
added the same logic as the projects in the articles, added info to th article.json and made a typography folder with reusable text elements

Commit:
finished most of the design and structure with(new): footer and updated styling on all sections, need to make minor styling fixes and look at accessibility and stretch goals

Commit:
case-ing bugfixes to ensure netlify deploy

Commit:
added the herotext back in

Commit:
debugging for netlify deploy

Commit:
Final: Ensure 'articles' directory is lowercase for Netlify

Commit:
Temp rename articles to force casing reset

Commit:
Step 1: Untrack the Articles folder

Commit:
Step 2: Re-add articles folder with correct lowercase casing

Commit:
app.jsx bugfix for deploy

Commit:
fixed styling for desktop, and started working on responsive styling for mobile and tablet

Commit:
mostly done with responsive design

Commit:
added netlify link in readme

Commit:
minor style fixes

Commit:
article mobile styling fix

Commit:
bugfix for mobile

Commit:
bugdetecting for netlify on phone

Commit:
more debugging for netlify

Commit:
flex-wrap for the project tags

Commit:
added 'see more projects' button

Commit:
added favicon

Commit:
added og tags to index.html

Commit:
look away unused socials links

Commit:
saved index.html

Npahlfer reviewed (approved):
✅

qabalany commented on src/App.jsx:
```diff
@@ -1,8 +1,33 @@
+import styled, { createGlobalStyle } from "styled-components"
+import { HeroSection } from "./components/hero/HeroSection"
+import { TechSection } from "./components/Tech/TechSection"
+import { ProjectSection } from "./components/projects/ProjectSection"
+import { SkillSection } from "./components/Skills/SkillSection"
+import { ArticleSection } from "./components/articles/ArticleSection"
+import { FooterSection } from "./components/footer/FooterSection"
```
Clean imports, nicely organized.

qabalany commented on src/App.jsx:
```diff
@@ -1,8 +1,33 @@
+}
+`
+
 export const App = () => {
```
Simple and readable component structure

qabalany commented on src/App.jsx:
```diff
@@ -1,8 +1,33 @@
+margin: 0;
+}
+`
+
```
GlobalStyles in the same file keeps things together nicely

qabalany commented on src/theme.jsx:
```diff
@@ -0,0 +1,13 @@
+export const theme = {
+
+  breakpoints: {
```
Breakpoints are well organized - mobile first approach is smart

qabalany commented on src/theme.jsx:
```diff
@@ -0,0 +1,13 @@
+    desktop: '1024px',
+  },
+
+  color: {
```
Centralizing colors makes future changes super easy

qabalany commented on src/theme.jsx:
```diff
@@ -0,0 +1,13 @@
+    primary: '#000',
+    secondary: '#FFF'
+  }
+}
```
Future idea:
As your project grows, you might enjoy adding spacing values here too. Would make your padding/margins consistent everywhere!

qabalany commented on src/main.jsx:
```diff
@@ -1,12 +1,15 @@
 import './index.css'
 
 createRoot(document.getElementById('root')).render(
   <StrictMode>
```
Perfect setup! StrictMode for catching issues, ThemeProvider wrapping everything

qabalany commented on src/components/hero/HeroSection.jsx:
```diff
@@ -0,0 +1,44 @@
+`
+
+
+const StyledHeroSection = styled.section`
```
Nice responsive padding that scales up with screen size

qabalany commented on src/components/hero/HeroSection.jsx:
```diff
@@ -0,0 +1,44 @@
+    <StyledHeroSection>
+      <HeroHeading></HeroHeading>
+      <StyledImageContainer>
+        <HeroImage position="left" src="/images/hero/SideProfile-heroImg-2.jpg" alt="side profile picture of carolina" />
```
Descriptive alt texts like "side profile picture of carolina" - great for accessibility!

qabalany commented on src/components/hero/HeroImage.jsx:
```diff
@@ -0,0 +1,65 @@
+import styled from "styled-components"
+
+const getPositionStyling = (position) => {
```
getPositionStyling function is clever! Clean way to handle different positions

qabalany commented on src/components/hero/HeroImage.jsx:
```diff
@@ -0,0 +1,65 @@
+const StyledImage = styled.img`
+  object-fit: cover;
+  border-radius: 12px;
+  ${props => getPositionStyling(props.$position)};
```
Using $position (transient prop) correctly - prevents prop leaking to DOM

qabalany commented on src/components/hero/HeroImage.jsx:
```diff
@@ -0,0 +1,65 @@
+        ? `margin-left: -160px; margin-top: 43px;`
+        : `margin: 16px 0;`}
+  }
+}
```
There's an extra closing brace } that looks like it might be a typo
The code works, but that extra } on line 60 seems unintended. Worth double-checking!

qabalany commented on src/components/projects/ProjectCard.jsx:
```diff
@@ -0,0 +1,85 @@
+`
+
+export const ProjectCard = ({ name, info, netlify, github, image, tags, index }) => {
+  const shouldReverse = index % 2 !== 0
```
index % 2 !== 0 for alternating layouts,  nice touch!

Npahlfer replied:
modulo, nice!

qabalany commented on src/components/projects/ProjectCard.jsx:
```diff
@@ -0,0 +1,85 @@
+      <ProjectCardImage src={image} alt={`Screenshot of ${name}`} />
+      <StyledLeftContent>
+        <StyledTagContainer>
+          {tags.map(tag => (
```
Mapping through tags with proper keys nice work

qabalany commented on src/components/projects/ProjectCard.jsx:
```diff
@@ -0,0 +1,85 @@
+        </StyledTagContainer>
+        <ProjectCardText name={name} info={info} />
+        <StyledButtonContainer>
+          <StyledButton
```
Small idea
The buttons look great! If you wanted, you could add target="_blank" for external links so they opens in new tab and adds security. Optional though!

```
<StyledButton
  href={netlify}
  text="Live Demo"
  icon="/icons/Netlify.svg"
  target="_blank"
  rel="noopener noreferrer"
/>
```

qabalany commented on src/components/projects/ProjectSection.jsx:
```diff
@@ -0,0 +1,38 @@
+import styled from "styled-components"
+import { ProjectHeading } from "./ProjectHeading"
+import { ProjectCard } from "./ProjectCard"
+import data from "../../data/projects.json"
```
Importing JSON data directly, clean approach

qabalany commented on src/components/projects/ProjectSection.jsx:
```diff
@@ -0,0 +1,38 @@
+    <>
+      <ProjectHeading />
+
+      {data.projects.map((project, index) => (
```
Nice prop spreading to ProjectCard

qabalany commented on src/components/assets/Button.jsx:
```diff
@@ -0,0 +1,34 @@
+import styled from "styled-components"
+
+const StyledButton = styled.a`
```
Well-styled, reusable button component

qabalany commented on src/components/assets/Button.jsx:
```diff
@@ -0,0 +1,34 @@
+    <>
+      <StyledButton href={href} className={className}>
+        <IconWrapper>
+          <img src={icon} alt="" />
```
Empty alt on decorative icon is correct accessibility practice!

qabalany commented on src/components/footer/FooterSection.jsx:
```diff
@@ -0,0 +1,37 @@
+import { FooterContactInfo } from "./FooterContactInfo"
+import { FooterSocials } from "./FooterSocials"
+
+const StyledFooter = styled.footer`
```
Nice centered layout with consistent spacing

qabalany commented on src/components/footer/FooterSocials.jsx:
```diff
@@ -0,0 +1,22 @@
+import styled from "styled-components"
+
+const SocialIconLink = styled.a`
```
Simple, focused component

qabalany commented on src/components/footer/FooterSocials.jsx:
```diff
@@ -0,0 +1,22 @@
+export const FooterSocials = (props) => {
+  return (
+    <>
+      <SocialIconLink href={props.href}>
+        <img src={props.src} alt={props.alt} />
+      </SocialIconLink>
+    </>
+  )
```
The fragment <> </> isn't needed here since you're returning one element

qabalany commented on index.html:
```diff
@@ -1,13 +1,88 @@
+
+<head>
+  <meta charset="UTF-8" />
+  <link
+    rel="icon"
+    type="image/x-icon"
+    href="/favicon/favicon.ico"
+  >
+  <link
+    rel="icon"
+    type="image/png"
+    sizes="32x32"
+    href="/favicon/favicon-32x32.png"
+  >
+  <link
+    rel="icon"
+    type="image/png"
+    sizes="16x16"
+    href="/favicon/favicon-16x16.png"
+  >
+
+  <link
+    rel="manifest"
+    href="/favicon/site.webmanifest"
+  >
+
+  <link
+    rel="apple-touch-icon"
+    sizes="180x180"
+    href="/favicon/apple-touch-icon.png"
+  >
```
Thorough favicon setup for all devices

qabalany commented on index.html:
```diff
@@ -1,13 +1,88 @@
+    href="https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&family=Montserrat:ital,wght@0,100..900;1,100..900&family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap"
+    rel="stylesheet"
+  >
+  <title>Portfolio</title>
```
Small SEO suggestion:
 The title is generic
```

Current 
<title>Portfolio</title>

More SEO-friendly
<title>Carolina Oldertz | Frontend Developer</title>

```

qabalany commented on src/data/projects.json:
```diff
@@ -1,28 +1,55 @@
-      "name": "Weather app",
-      "image": "https://images.unsplash.com/photo-1520792532857-293bd046307a?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2370&q=80",
+      "name": "RECIPE LIBRARY",
+      "info": "Using spoonaculars API I have fetched a library of recipes that displays ingredients, cooking time and meal type. You can filter and sort by different categories and search for your recipe and aslo with a button click to geta random recipe.",
```
Tiny typo
"aslo" should be "also"
Also "geta" should be "get a" 😊

qabalany reviewed (commented):
🎉
Hey! I went through your portfolio and it's looking great! 

This is really solid work! The code is clean, organized, and shows you understand React well. The responsive design is thorough, and little touches like proper alt texts show attention to detail.

The suggestions above are all minor polish - nothing is broken, and the portfolio looks professional. Keep up the great work! 🚀

P.S. - Love the cat subscription site with party mode. That creativity really shows your personality! 🐱

Npahlfer commented on src/components/typography/H1.jsx:
```diff
@@ -0,0 +1,18 @@
+import styled from "styled-components"
```
Instead of defining a set of different Typographies like you have done in here, it would be nicer to define a component that can change (both the styling and the element) depending on the prop. Like we did in class:
https://github.com/Technigo/fall-2025-live-sessions/blob/main/week-14/styled-components/src/components/Typography.jsx

Npahlfer commented on src/index.css:
This file can be deleted as you have global styling in the GlobalStyle function and the rest in each component :)

Npahlfer commented on src/theme.jsx:
```diff
@@ -0,0 +1,13 @@
+export const theme = {
```
You could even expand with more themes here eg. spacing and font styling.

Npahlfer commented on src/components/projects/ProjectCardImage.jsx:
```diff
@@ -0,0 +1,21 @@
+  object-fit: cover;
+  border-radius: 12px;
+  box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.25);
+  object-position: ${props => (props.$altText === 'Screenshot of RECIPE LIBRARY' ? 'left' : 'center')};
```
Checking for an alt text like this is a bad idea because the alt text might change and then this doesnt work anymore. Better to use another prop on the ProjectCardImage (eg. direction="left")

Npahlfer commented on src/components/Skills/SkillSection.jsx:
```diff
@@ -0,0 +1,78 @@
+  },
+  {
+    tag: "Toolbox",
+    info: "Atom\nPostman\nAdobe Photoshop\nAdobe Illustrator\nAdobe InDesign\nAdobe Acrobat\nAdobe Premiere Pro & Rush\nAdobe After Effects\nCamera raw\nBridge\nWordpress\nSitecore (CMS)\nSitevison (CMS)\nShopify\nFigma\nMicrosoft (Word, ppt, etc)\nSlack",
```
instead of having them separated with a linebreak, use an array! :)

Npahlfer replied:
then you dont have to use the split like you do later in SkillInfo

Npahlfer commented:
There are some room for improvements (check my comments) like the expansion of the theme and some logic parts, but overall really good! :)

Commit:
fixed small typos and seo, changed a recipe photo logic and changed typography to be one component, (all according to feedback)

carro-barro committed:
Enhance README with project details and setup instructions

Added detailed project description, key features, tech stack, project structure, and setup instructions.
